test/gurka/test_build_admin.cc

```diff
@@ -0,0 +1,128 @@
+#include <gtest/gtest.h>
+
+#include "gurka.h"
+
+#include "baldr/admin.h"
+#include "mjolnir/admin.h"
+#include "mjolnir/adminbuilder.h"
+#include "mjolnir/pbfgraphparser.h"
+
+using namespace valhalla;
+using namespace valhalla::baldr;
+using namespace valhalla::mjolnir;
+
+class AdminTest : public ::testing::Test {
+protected:
+  static gurka::map admin_map;
+
+  static void SetUpTestSuite() {
+    constexpr double gridsize = 10;
+
+    const std::string ascii_map = R"(
+          A-------B-------C
+          |       |       |
+          |       |       |
+          |       |       |
+          F-------E-------D
+    )";
+
+    // To define an administrative boundary, the nodes must form a closed polygon.
+    const gurka::ways ways = {{"ABCDEFA",
+                               {
+                                   {"highway", "primary"},
+                               }},
+                              {"ABEFA",
+                               {
+                                   {"highway", "primary"},
+                               }},
+                              {"BCDEB",
+                               {
+                                   {"highway", "primary"},
+                               }}};
+
+    const gurka::relations relations = {{{{{gurka::way_member, "ABEFA", "outer"}}},
+                                         {
+                                             {"type", "boundary"},
+                                             {"boundary", "administrative"},
+                                             {"admin_level", "4"},
+                                         }},
+                                        {
+                                            {{{gurka::way_member, "BCDEB", "outer"}}},
+                                            {
+                                                {"type", "boundary"},
+                                                {"boundary", "administrative"},
+                                                {"admin_level", "4"},
+                                            },
+                                        },
+                                        {{{{gurka::way_member, "ABCDEFA", "outer"}}},
+                                         {
+                                             {"type", "boundary"},
+                                             {"boundary", "administrative"},
+                                             {"admin_level", "2"},
+                                         }}};
+
+    const auto layout = gurka::detail::map_to_coordinates(ascii_map, gridsize);
+    admin_map = gurka::buildtiles(layout, ways, {}, relations, "test/data/admin");
```

Comment:
i hadnt thought about this before but its quite odd to have to actually build tiles just so you can get the pbf written. can we expose the methods in gurka to write just the pbf to file so you can reference that in the test below?

Reply:
I have no idea.  Lemme see.

Reply:
Ok, I tried to remove the logic that builds the tiles - only to find out I actually need the tile data.  GetAdminInfo() requires the caller to provide a GraphTileBuilder, whose constructor requires me to pass in a folder to tile data.